prophyc/parsers/sack.py

```diff
@@ -115,15 +117,21 @@ def add_union(self, cursor):
         node = model.Union(alphanumeric_name(cursor), members)
         self._add_node(node)
 
-def build_model(tu):
-    builder = Builder()
+def build_model(tu, isar_supples=[]):
+    builder = Builder(isar_supples)
     for cursor in tu.cursor.get_children():
         if cursor.kind is CursorKind.UNEXPOSED_DECL:
             for in_cursor in cursor.get_children():
                 if in_cursor.kind is CursorKind.STRUCT_DECL and in_cursor.spelling and in_cursor.is_definition():
                     builder.add_struct(in_cursor)
-        if cursor.kind is CursorKind.STRUCT_DECL and cursor.spelling and cursor.is_definition():
-            builder.add_struct(cursor)
+        if cursor.spelling and cursor.is_definition():
+            if cursor.kind is CursorKind.STRUCT_DECL:
+                builder.add_struct(cursor)
+            if cursor.kind is CursorKind.ENUM_DECL:
+                builder.add_enum(cursor)
+    if isar_supples:
+        cheated_names = list(get_node_names(isar_supples))
+        builder.nodes = [n for n in builder.nodes if n.name not in cheated_names]
```

Comment:
This probably wouldn't be needed if clang didn't read cheat-generated c++ files. Getting unknown type name from cursor.extent would allow to consult includes in search for node, not generating duplicated struct to filter out.